coffee/base.py

```diff
@@ -243,25 +280,27 @@ class ColSparseArrayInit(ArrayInit):
     """Array initilizer in which zero-columns, i.e. columns full of zeros, are
     explictly tracked. Only bi-dimensional arrays are allowed."""
 
-    def __init__(self, values, nonzero_bounds, numpy_values):
-        """Zero columns are tracked once the object is instantiated.
-
-        :arg values: string representation of the values the array is initialized to
-        :arg zerobounds: a tuple of two integers indicating the indices of the first
-                         and last nonzero columns
+    def __init__(self, values, precision, nonzero_bounds):
+        """Initialize an ArrayInit object in which blocks of zero columns are
+        explicitly tracked.
+
+        :arg values: representation of the values the array is initialized to
+        :type values: a string or a numpy ndarray.
+        :arg precision: the number of decimal digits that should be used when
+            converting a float (in a numpy array) to a string.
+        :type precision: integer (defaults to 12)
+        :arg zerobounds: a list of 2-tuple of integers, each tuple indicating
+                         the indices of the columns at the boundary of a dense
+                         block
         """
-        super(ColSparseArrayInit, self).__init__(values)
+        super(ColSparseArrayInit, self).__init__(values, precision)
         self.nonzero_bounds = nonzero_bounds
-        self.numpy_values = numpy_values
 
-    def reconstruct(self, values, nonzero_bounds, numpy_values, **kwargs):
-        return type(self)(values, nonzero_bounds, numpy_values, **kwargs)
+    def reconstruct(self, values, precision, nonzero_bounds, **kwargs):
+        return type(self)(values, precision, nonzero_bounds, **kwargs)
 
     def operands(self):
-        return [self.values, self.nonzero_bounds, self.numpy_values], {}
-
-    def gencode(self, not_scope=True, parent=None):
-        return self.values
+        return [self.values, self.nonzero_bounds], {}
```

Comment:
This can't be the correct argument list to return from operands, because where has the precision gone?